src/main/functions/SIOPVersionDiscovery.ts

```diff
@@ -0,0 +1,36 @@
+import Ajv from 'ajv';
+
+import { D11AuthenticationRequestPayloadSchema, ID1AuthenticationRequestPayloadSchema } from '../schemas';
+import { AuthenticationRequestPayload, SupportedVersion } from '../types';
+
+// Need to remove the JWT fields to verify the request against the schema
+function removeJWTPropertiesFromAuthRequest(authenticationRequestPayload: AuthenticationRequestPayload) {
+  delete authenticationRequestPayload['iat'];
```

Comment:
I thought it would make sense to verify everything in the authentication request in a single place, however looking at it from the SIOP spec version perspective, I just need to add a method to the OP with accepts the authentication request payload and checks the version. Fixed